pkg/asset/installconfig/gcp/validation.go

```diff
@@ -336,3 +337,48 @@ func validateCredentialMode(client API, ic *types.InstallConfig) field.ErrorList
 
 	return allErrs
 }
+
+func validateMarketplaceImages(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	const errorMessage string = "could not find the boot image: %v"
+	const unspecifiedArch string = "ARCHITECTURE_UNSPECIFIED"
+
+	if ic.GCP.DefaultMachinePlatform != nil && ic.GCP.DefaultMachinePlatform.OSImage.Name != "" {
+		osImage := ic.GCP.DefaultMachinePlatform.OSImage
+		image, err := client.GetImage(context.TODO(), osImage.Name, osImage.Project)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "gcp", "defaultMachinePlatform", "osImage"), osImage, fmt.Sprintf(errorMessage, err)))
+		} else if image != nil && image.Architecture == unspecifiedArch {
+			logrus.Warn(fmt.Sprintf("Boot image architecture is unspecified and might not be compatible with %s controlPlane nodes", ic.ControlPlane.Architecture))
+		}
+	}
+
+	if ic.ControlPlane != nil && ic.ControlPlane.Platform.GCP != nil && ic.ControlPlane.Platform.GCP.OSImage.Name != "" {
+		osImage := ic.ControlPlane.Platform.GCP.OSImage
+		image, err := client.GetImage(context.TODO(), osImage.Name, osImage.Project)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("controlPlane", "platform", "gcp", "osImage"), osImage, fmt.Sprintf(errorMessage, err)))
+		} else if image != nil && string(image.Architecture) == unspecifiedArch {
+			logrus.Warn(fmt.Sprintf("Boot image architecture is unspecified and might not be compatible with %s controlPlane nodes", ic.ControlPlane.Architecture))
+		}
+	}
+
+	for idx, compute := range ic.Compute {
+		if compute.Platform.GCP == nil {
+			continue
+		}
+		fieldPath := field.NewPath("compute").Index(idx)
+		osImage := compute.Platform.GCP.OSImage
+		if osImage.Name != "" {
+			image, err := client.GetImage(context.TODO(), osImage.Name, osImage.Project)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("platform", "gcp", "osImage"), osImage, fmt.Sprintf(errorMessage, err)))
+			} else if image != nil && image.Architecture == unspecifiedArch {
+				logrus.Warn(fmt.Sprintf("Boot image architecture is unspecified and might not be compatible with %s compute nodes", ic.ControlPlane.Architecture))
+			}
```

Comment:
Missing check for `image.Architecture` != `compute.Architecture`. See comments https://github.com/openshift/installer/pull/7215#discussion_r1222050996